agent/background.go

```diff
@@ -68,7 +68,7 @@ func (a *Agent) startIdleTimeoutWatch(ctx context.Context, tc *taskContext, time
 				close(timeout)
 				return
 			}
-			go a.updateIdleTimeout(ctx, tc, nextTimeout, resetIdleTimeout)
+			a.updateIdleTimeout(ctx, tc, nextTimeout, resetIdleTimeout)
```

Comment:
This won't work because the same select statement is receiving on line 72 what `a.updateIdleTimeout` sends on line 71, so it will always find itself blocked and never reset the idle timeout (will also hit the default case on line 128 of agent/task.go).